### PR TITLE
[com_actionlogs] - log actionlog purge/export

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #22695.

### Summary of Changes

- row/event is inserted/triggred after the purge
- row/event is inserted/triggred after the export

### Testing Instructions
Export/Purge the action logs
see  #22695

### CODE REVIEW

**Comment by PhilETaylor** [comment]
Does the export language need to reflect if EVERYTHING was exported, or if just a few rows were exported?

File: plugins/actionlog/joomla/joomla.php [hunk 1]
```diff
@@ -907,4 +907,64 @@ public function onUserAfterRemind($user)
 
 		$this->addLog(array($message), 'PLG_ACTIONLOG_JOOMLA_USER_REMIND', $context, $user->id);
 	}
+
+	/**
+	 * On after Cache purge
```
Comment (brianteeman) [inline plugins/actionlog/joomla/joomla.php]:
```suggestion
	 * On after log action purge
```

**Review by brianteeman — APPROVED** [review]
approve language changes

File: administrator/components/com_actionlogs/models/actionlogs.php [hunk 2]
```diff
@@ -312,13 +314,15 @@ public function purge()
...
 			$this->getDbo()->truncateTable('#__action_logs');
-
-			return true;
 		}
 		catch (Exception $e)
 		{
 			return false;
 		}
+
+		JFactory::getApplication()->triggerEvent('onAfterLogPurge', array());
```
Comment (laoneo) [inline administrator/components/com_actionlogs/models/actionlogs.php]: Just a detail. But would be good if you can use the namespaced classes.

Comment (alikon) [reply administrator/components/com_actionlogs/models/actionlogs.php]: done

File: administrator/components/com_actionlogs/controllers/actionlogs.php [hunk 3]
```diff
@@ -97,7 +97,8 @@ public function exportLogs()
 			}
 
 			fclose($output);
-
+
```
Comment (Quy) [inline administrator/components/com_actionlogs/controllers/actionlogs.php]:
```suggestion

```

File: administrator/components/com_actionlogs/models/actionlogs.php (hunk 2 above)
Comment (SharkyKZ) [reply administrator/components/com_actionlogs/models/actionlogs.php]: Import `Joomla\CMS\Factory` or revert to aliased class.

**Comment by ReLater** [comment]
I have tested this item :red_circle: unsuccessfully on a4b8b28aa0078edbf7c8321b7cf291ca361f749d<br /><br />Class 'Factory' not found
/administrator/components/com_actionlogs/models/actionlogs.php:300 

Class 'Factory' not found
/administrator/components/com_actionlogs/models/actionlogs.php:323 <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22740">issues.joomla.org/tracker/joomla-cms/22740</a>.</sub>

**Comment by alikon** [comment]
@ReLater please retest

**Comment by ReLater** [comment]
I have tested this item :white_check_mark: successfully on 7f60480cdd5eba4835abd707d9a9b126617e9232<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22740">issues.joomla.org/tracker/joomla-cms/22740</a>.</sub>

**Comment by PhilETaylor** [comment]
> Does the export language need to reflect if EVERYTHING was exported, or if just a few rows were exported?

What happens when a few rows are deleted? nothing? if we are rewriting history then that should be logged too right?

**Comment by alikon** [comment]
the `Factory::getApplication()->triggerEvent('onAfterLogPurge', array());`

happens
:smile: 

https://github.com/joomla/joomla-cms/pull/22740/files#diff-6d4ae54d455c6ccba7a75cc85b8bca19R301
https://github.com/joomla/joomla-cms/pull/22740/files#diff-6d4ae54d455c6ccba7a75cc85b8bca19R324

so no, no different messages are logged if you delete some or all

**Comment by PhilETaylor** [comment]
so in that case saying

> User <a href='{accountlink}'>{username}</a> purged the action log

is actually factually incorrect and the language/terminology used needs to be changed - maybe something like

> User <a href='{accountlink}'>{username}</a> purged one or more rows from the action log

**Comment by alikon** [comment]
feel free to submit the correct string, pr is open
english is not my best friend ;)

**Comment by PhilETaylor** [comment]
As the [only codeowner for language strings](https://github.com/joomla/joomla-cms/blob/staging/.github/CODEOWNERS), the ultimate decision is with @brianteeman - he has already approved your language string even though it is factually incorrect. 

I suggest he writes the factually correct sentence using his English skills.

**Comment by tecpromotion** [comment]
I have tested this item :white_check_mark: successfully on a9da4b805729b1f2dbb266c51e065a62fc7ce84a<br /><br />I have tested this successfully on Joomla! 3.9.2.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22740">issues.joomla.org/tracker/joomla-cms/22740</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on a9da4b805729b1f2dbb266c51e065a62fc7ce84a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22740">issues.joomla.org/tracker/joomla-cms/22740</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22740">issues.joomla.org/tracker/joomla-cms/22740</a>.</sub>

**Comment by HLeithner** [comment]
thx

**Pull request merged**